amr-wind/physics/SyntheticTurbulence.H

```diff
@@ -0,0 +1,143 @@
+#ifndef SyntheticTurbulence_H
+#define SyntheticTurbulence_H
+
+#include <string>
+#include <cmath>
+#include <memory>
+
+#include "amr-wind/core/Physics.H"
+#include "amr-wind/core/Field.H"
+#include "amr-wind/core/vs/vector_space.H"
+#include "amr-wind/CFDSim.H"
+
+namespace amr_wind {
+
+namespace synth_turb {
+class MeanProfile
+{
+public:
+  MeanProfile(double ref_vel, int shear_dir=2)
+    : m_ref_vel(ref_vel),
+      m_shear_dir(shear_dir)
+  {}
+
+  virtual ~MeanProfile() = default;
+
+  virtual double operator()(double /* height */) const
+  { return m_ref_vel; }
+
+  inline double reference_velocity() const { return m_ref_vel; }
+
+  inline int shear_dir() const { return m_shear_dir; }
+
+protected:
+  const double m_ref_vel;
+  const int m_shear_dir;
+};
+
+} // namespace synth_turb
+
+struct SynthTurbData
+{
+  // Dimensions of the box
+  vs::VectorT<int> box_dims;
+
+  // Length of the boxes in each direction
+  vs::Vector box_len;
+
+  vs::Vector dx;
+
+  // Reference point for the turbulence box. Reference is the mid-point of the
+  // turbulence grid at the plane where it is injected into the CFD flow field.
+  vs::Vector origin;
+
+  // Transformation matrix to convert from global coordinate system to local
+  // coordinate system
+  vs::Tensor tr_mat;
+
+  // Perturbation velocities (2, ny, nz)
+  std::vector<double> uvel;
+  std::vector<double> vvel;
+  std::vector<double> wvel;
+
+  // Indices of the two planes stored in the data arrays
+  int ileft;
+  int iright;
+};
+
+
+
+/**
+ *  \defgroup synth_turb Synthetic Turbulence
+ *
+ *  SyntheticTurbulence contains functions to inject turbulence into a
+ *  CFD simulation. It reads a file to get the turbulent velocity on a
+ *  grid and populates a source term for the momentum equation to
+ *  achieve the desired turbulence characteristics at a specified
+ *  plane in the CFD domain.
+ *
+ * \ingroup physics
+ */
+
+class SyntheticTurbulence: public Physics::Register<SyntheticTurbulence>
+{
+public:
+
+  static const std::string identifier() { return "SyntheticTurbulence"; }
+
+  struct NCBoxTurb
+  {
+    std::string filename;
+  };
```

Comment:
Not needed anymore?

Reply:
Fixed in https://github.com/Exawind/amr-wind/pull/327/commits/5213aab7053bb736e11a39c02ff753c2820f8837